### ISSUE TITLE
Create minimal DQN training script

### ISSUE THREAD

**Issue by CooperBigFoot** [CooperBigFoot/RL-demo#7]
## Description
Implement the main training loop using TorchRL's SyncDataCollector and training patterns.

## Reference
See planning document: `/scratchpads/phase3_integration_plan.md` - Sections 1, 2

## Dependencies
- #3 (Environment)
- #5 (Q-network)
- #6 (DQN components)

## Tasks
- [ ] Create `scripts/train_dqn.py`
- [ ] Implement training loop:
  - Create environment
  - Initialize Q-network and target network
  - Setup DQNLoss and optimizer
  - Configure SyncDataCollector
  - Main training loop with replay buffer
- [ ] Add basic logging:
  - Episode rewards
  - Loss values
  - Epsilon decay
- [ ] Add checkpointing:
  - Save every 5000 steps
  - Save best model
- [ ] Create simple configuration:
  - Learning rate: 1e-3
  - Batch size: 32
  - Total frames: 50,000 (for initial testing)

## Code Structure
```python
from torchrl.collectors import SyncDataCollector
from torchrl.data import ReplayBuffer, LazyTensorStorage

def main():
    # Create environment
    env = DiscreteReservoirEnv(config)
    
    # Create networks
    qnet = create_qnet()
    target_qnet = create_qnet()
    
    # Create loss and optimizer
    loss_module = create_dqn_loss(qnet)
    optimizer = torch.optim.Adam(qnet.parameters(), lr=1e-3)
    
    # Create collector
    collector = SyncDataCollector(
        env,
        policy=qnet,
        frames_per_batch=256,
        total_frames=50000,
    )
    
    # Training loop
    for batch in collector:
        # Add to replay buffer
        # Sample and train
        # Update target network
        # Log metrics
```

## Acceptance Criteria
- Training runs without errors
- Loss decreases over time
- Agent shows learning progress
- Checkpoints are saved correctly

## Priority
High - Main training implementation
